dspace-iiif/src/main/java/org/dspace/app/iiif/service/AnnotationListService.java

```diff
@@ -60,6 +63,37 @@ public AnnotationListService(ConfigurationService configurationService) {
         setConfiguration(configurationService);
     }
 
+
+    /**
+     * Returns image annotation list from the bitstream metadata field. The
+     * JSON annotation list is created outside DSpace and added to bitstream
+     * metadata by the user.
+     * @param context DSpace context
+     * @param uuid bitstream UUID
+     * @return IIIF annotation list
+     */
+    public String getImageAnnotations(Context context, UUID uuid) throws RuntimeException {
+        BitstreamService bitstreamService = ContentServiceFactory.getInstance().getBitstreamService();
+        Bitstream bitstream = null;
+        try {
+            bitstream = bitstreamService.find(context, uuid);
+            if (bitstream == null) {
+                throw new ResourceNotFoundException("DSpace bitstream for  id " + uuid + " not found");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        List<MetadataValue> metadata = bitstreamService.getMetadata(bitstream, "iiif", "media",
+            "annotations", Item.ANY);
+
+        // If annotations exist, the JSON formatted list will be found in the first, non-repeating metadata field.
+        if (metadata.size() > 0) {
+            return metadata.get(0).getValue();
+        } else {
+            throw new ResourceNotFoundException("IIIF Annotation List not found for  id " + uuid);
```

Comment:
As an "example/simple" implementation I would suggest to use a dedicated bundle for annotation and set a rule like the one used for thumbnail to associate the annotation bitstream with the annotated one.
Alternatively we can allow to store the full URL to retrieve the annotation in the image bitstream metadata and use this information to trigger a redirect from the internal dspace annotation url

Reply:
Hi @abollini , I like the idea of the dedicated bundle for annotations. I just pushed an implementation of that approach.